app/controllers/secret_shares_controller.rb

```diff
@@ -0,0 +1,9 @@
+class SecretSharesController < ApplicationController
+
+  def create
+    moment = Moment.friendly.find(params[:moments])
+    moment.update :secret_share_identifier => SecureRandom.uuid
```

Comment:
Thanks for the feedback. I have looked into update vs update! I have made the change to the controller and committed the change. I am keen to learn more about your thoughts about moving into the model and adding a uniqueness validator.